edn/src/types.rs

```diff
@@ -87,15 +87,57 @@ fn test_print_edn() {
                Value::Boolean(true))).to_string());
 }
 
-impl Value {
-    pub fn is_keyword(&self) -> bool {
-        match *self {
-            Keyword(_) => true,
-            _          => false,
+macro_rules! def_is {
```

Comment:
In general macros are a little hard to read, so a `///` comment explaining what they try to achieve goes a long way.

Also, a nit, consider `$pat` instead of `$kind`; _kind_ [means something precise in type theory](https://en.wikipedia.org/wiki/Kind_(type_theory)) that does not agree with your usage here.